server/src/test/java/org/opensearch/indices/recovery/LocalStorePeerRecoverySourceHandlerTests.java

```diff
@@ -746,6 +749,104 @@ void phase2(
         assertFalse(phase2Called.get());
     }
 
+    public void testThrowExceptionOnNoTargetInRouting() throws IOException {
+        final RecoverySettings recoverySettings = new RecoverySettings(Settings.EMPTY, service);
```

Comment:
Should we have a test around the case where after 2-3 retries the routing is found?